Shared/Samples/Create and save KML file/CreateAndSaveKMLView.Model.swift

```diff
@@ -0,0 +1,135 @@
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Combine
+
+extension CreateAndSaveKMLView {
+    /// The model used to store the geo model and other expensive objects used in this view.
+    @MainActor class Model: ObservableObject {
+        /// A dark gray map.
+        let map = Map(basemapStyle: .arcGISDarkGray)
+        
+        /// The geometry editor that deals with the sketching.
+        let geometryEditor: GeometryEditor
+        
+        /// The KML style used for the geometry.
+        var kmlStyle: KMLStyle?
+        
+        /// A KML document that serves as a container for features and styles.
+        var kmlDocument = KMLDocument()
+        
+        /// A KMZ file that can be exported with the system's file exporter.
+        var kmzFile = KMZFile(document: .init())
+        
+        /// A Boolean value indicating if the geometry can be cleared from the geometry editor.
+        @Published private(set) var canClearCurrentSketch = false
+        
+        /// A Boolean value indicating if the saved sketches can be cleared.
+        @Published private(set) var canClearSavedSketches = false
+        
+        /// A Boolean value indicating if the geometry editor has started.
+        @Published var isStarted = false
+        
+        /// A Boolean value indicating if the geometry can be saved to a graphics overlay.
+        @Published private(set) var canSave = false
+        
+        /// The current geometry of the geometry editor.
+        @Published private(set) var geometry: Geometry? {
+            didSet {
+                canClearCurrentSketch = geometry.map { !$0.isEmpty } ?? false
+                canSave = geometry?.sketchIsValid ?? false
+            }
+        }
+        
+        /// A Boolean value indicating if we should show the file exporter.
+        @Published var showingFileExporter = false
+        
+        /// A Boolean value indicating if the file exporter button should be disabled.
+        @Published var fileExporterButtonIsDisabled = true
+        
+        /// A Boolean value indicating whether to show an alert.
+        @Published var isShowingAlert = false
+        
+        /// The error shown in the alert.
+        @Published var error: Error? {
+            didSet { isShowingAlert = error != nil }
+        }
+        
+        /// Creates the model for this view.
+        init() {
+            self.geometryEditor = GeometryEditor()
+            
+            resetKMLLayer()
+            
+            Task { [weak self, geometryEditor] in
```

Comment:
https://github.com/Esri/arcgis-maps-sdk-swift-samples/issues/153

Can you examine if this causes a retain cycle? Thanks.

Reply:
I added a deinit with a print statement in the model to make sure it was getting released and it looks like it was but I moved the async work into a task view modifier anyway so the user knows the preferred way of doing it.

Reply:
>  or (preferred) use a task modifier on the view itself.

Yep that is the approach I took, thank you.